### DISCUSSION TITLE
Is there a way to predefine constraint size?

### DISCUSSION THREAD

**Discussion by tsofrank** [yalmip/YALMIP#1433] [General discussion]
Hi community,
I am using yalmip in in MATLAB to solve an optimization problem with semi-definite constraints.
I currently define constraints in a for loop by 

> for i = 1:M 
...
         constr = [constr; lmiOut(...) <= eps*eye(dimLMI)];
...
end

But my for loop are tens of thousands. Instead of changing the size of constr every time, is there a way I can predefine a list of a certain size and then assign constr(i)  in a for loop?
I know MATLAB r2024 has a function createArray, but it requires a *"classname"* as one of the arguments. If I try 
> constr = createArray(M,1,"lmi") 

or
> constr = createArray(M,1,"constraint")

I get an error messages
> Default constructor for class 'lmi' does not return a scalar object. /
Default constructor for class 'constraint' does not return a scalar object. Caused by:
    Not enough input arguments.

Does anyone have suggestions on how I might be able to predefine a fixed size of constraints? 
Many thanks,

Frank

**Comment by johanlofberg** [comment]
No, SDPs are not vectorizable (and YALMIP is storing stuff in split up lists in various ways to try to avoid some overhead when adding a single constraint to a list of constraints)
